### ISSUE TITLE
Legacy printing mode gives not-legacy result for array with negative value, value > 9

### ISSUE THREAD

**Issue by matthew-brett** [numpy/numpy#10383]
Numpy 1.13:

```
$ python -c 'import numpy; print(numpy.array([  -1.  ,    10]))'
[ -1.  10.]
```

There is whitespace before the`-1`.

Numpy 1.14:

```
$ python -c 'import numpy; print(numpy.array([  -1.  ,    10]))'
[-1. 10.]
$ python -c 'import numpy; numpy.set_printoptions(legacy="1.13"); print(numpy.array([  -1.  ,    10]))'
[-1. 10.]
```

No whitespace before the `-1`.

**Comment by KelSolaar** [comment]
This is breaking a fair bunch of my doctests: https://travis-ci.org/colour-science/colour/builds/328061368

**Comment by matthew-brett** [comment]
And mine :) - and we now can't fix them by adding something like this:

https://github.com/matthew-brett/nb2plots/commit/b98975f4e724684650547a9f40c91fcec9add007

**Comment by ahaldane** [comment]
Ah, the legacy case was stranger than I thought! 

Added a fix in #10361.
